From f216608d16f9d43fef089f67d0c693c2c781974d Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Tue, 22 Mar 2022 09:24:23 +0200
Subject: [PATCH] [SW-0] greco: init spsram only on cold rst

This patch add check to initialize spsram only when reset cause is cold
reset.

Change-Id: I0592e98046687d5a9772d542d3244cad223b6431
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/arc/common.c b/arch/greco/arc/common.c
index b77ceb7..bf642e6 100644
--- a/arch/greco/arc/common.c
+++ b/arch/greco/arc/common.c
@@ -178,9 +178,13 @@
 #define SP_SRAM_60K_SZ_16B_GRANULARITY	0xF00
 void preboot_sp_sram_init(void)
 {
-	u32 mem_cfg, cpy_sts;
+	u32 mem_cfg, cpy_sts, rst_src;
 	u64 start;
 
+	rst_src = hl_rst_src_get();
+	if (rst_src != HL_COLD_RST)
+		return;
+
 	/* Use memblock to initialize scratchpad memory from 4K offset
 	 * as first 4K offset is used by arcboot rom
 	 */
